### ISSUE TITLE
Helm-based HA install w/cni, viz, and auto-rotating cert-manager issuer cert fails linkerd-identity-data-plane check

### ISSUE THREAD

**Issue by ron1** [linkerd/linkerd2#7484]
### What is the issue?

I performed a Helm-based HA install w/cni, viz, and auto-rotating cert-manager issuer cert using the Rancher Fleet GitOps tool. While all pods seem to be running fine, the linkerd data plane appears to be non-functional as evidenced by the following output snippet from the `linkerd check --proxy` command:
```
linkerd-identity-data-plane
------------------------------
(yellow)  data plane proxies certificate match CA
    Some pods do not have the current trust bundle and must be restarted:
      * <list of all my custom meshed pods>
      * <list of all linkerd-viz pods>

linkerd-viz
-----------
...
(red) viz extension proxies are healthy
  Some pods do not have the current trust bundle and must be restarted:
    * <list of all linkerd-viz pods>

Status check results are (red)
```

Note that a standard, non-HA, non-cert-manager, cli-based install worked previously on the same cluster without a problem. I did completely uninstall this previous cli-based install before attempting this new, helm-based installation.

### How can it be reproduced?

1. Create linkerd-trust-anchor Secret and linkerd-identity-trust-roots ConfigMap in a common namespace
2. Configure kyverno v1.5.2 to mutate linkerd, linkerd-cni, and linkerd-viz namespaces with annotations and labels defined in namespace manifests in corresponding linkerd chart templates
3. Configure kyverno v1.5.2 to sync linkerd-trust-anchor Secret and linkerd-identity-trust-roots ConfigMap from common namespace to linkerd namespace
4. Install `linkerd-identity-issuer-cert.yaml` and `linkerd-trust-anchor-issuer.yaml` via kustomize
5. Install linkerd2-cni-2.11.1 chart with the following config:
```
values:
  installNamespace: false
```
6. Install linkerd2-2.11.1 chart with the following config:
```
values:
  cniEnabled: true
  identity:
    externalCA: true
    issuer:
      scheme: "kubernetes.io/tls"
  installNamespace: false
  nodeSelector: {"kubernetes.io/os":"linux"}

valuesFiles:
- values-ha.yaml
```
7. Install linkerd-viz-2.11.1 chart with the following config:
```
values:
  defaultRegistry: ghcr.io/linkerd
  grafana:
    enabled: true
  installNamespace: false
  nodeSelector: {"kubernetes.io/os":"linux"}
  prometheus:
    enabled: true
```

### Logs, error output, etc

See output above.

### output of `linkerd check -o short`

Airgapped deployment seems to skew these results
```
issuer cert is valid for at least 60 days
unsupported version channel: stable-2.11.1
some proxies are not running the current version: <list of all linkerd pods>
linkerd-viz - some pods do not have the current trust bundle and must be restarted: <list of all linkerd-viz pods>
```

### Environment

Kubernetes version: RKE2 1.21.5+rke2r1

### Possible solution

It may be that this issue is a duplicate of issue https://github.com/linkerd/linkerd2/issues/7449.

**Comment by mateiidavid** [comment]
Hey @ron1, just to confirm, does this affect the control plane pods or the proxies in any way? Does traffic break, or are pods stuck in initialisation? 

I think this might be a duplicate of https://github.com/linkerd/linkerd2/issues/7449 (as you noted in the issue description). We have a fix in-flight, let's keep this open for a while and see if the fix resolves it. As long as the only thing that fails is the check, this should be benign.

**Comment by ron1** [comment]
Just the healthcheck pods are in Error state. I need to confirm the control plane and data plane are fully functional otherwise.

Unfortunately the failing check seems to prevent use of the dashboard which is problematic for us. I am looking forward to verifying the fix that is in flight.

**Comment by adleong** [comment]
The above fix has been released in edge-21.12.4.  Can you try this version and see if it resolves the issue?

**Comment by ron1** [comment]
@adleong I am planning to verify this fix in edge-21.12.4 later this week. I will report results here as soon as I have them.Thanks.

**Comment by ron1** [comment]
Fixed in 22.1.1. Thanks.
